x-pack/agent/pkg/agent/operation/operator_handlers.go

```diff
@@ -67,15 +67,24 @@ func getProgramFromStepWithTags(step configrequest.Step, tags map[app.Tag]string
 }
 
 func getConfigFromStep(step configrequest.Step) (map[string]interface{}, error) {
-	metConfig, ok := step.Meta[configrequest.MetaConfigKey]
-	if !ok {
+	metConfig, hasConfig := step.Meta[configrequest.MetaConfigKey]
+
+	if !hasConfig && needsMetaConfig(step) {
 		return nil, fmt.Errorf("step: %s, no config in metadata", step.ID)
 	}
 
-	config, ok := metConfig.(map[string]interface{})
-	if !ok {
-		return nil, fmt.Errorf("step: %s, program config is in invalid format", step.ID)
+	var config map[string]interface{}
+	if hasConfig {
+		var ok bool
+		config, ok = metConfig.(map[string]interface{})
+		if !ok {
+			return nil, fmt.Errorf("step: %s, program config is in invalid format", step.ID)
+		}
 	}
 
 	return config, nil
 }
+
+func needsMetaConfig(step configrequest.Step) bool {
+	return step.ID == configrequest.StepRun || step.ID == configrequest.StepStartSidecar
+}
```

Comment:
Not use I understand the purpose of `StepStartSidecar` ?

Reply:
previous intended design was a bit different i will remove this